src/Simplex/Chat/Store.hs

```diff
@@ -1284,55 +1298,117 @@ updateConnectionStatus db Connection {connId} connStatus = do
+createGroupInvitation :: DB.Connection -> User -> Contact -> GroupInvitation -> Maybe Profile -> ExceptT StoreError IO GroupInfo
+createGroupInvitation db user@User {userId} contact@Contact {contactId} GroupInvitation {fromMember, fromMemberProfile, invitedMember, connRequest, groupProfile} incognitoProfile = do
   liftIO getInvitationGroupId_ >>= \case
-    Nothing -> ExceptT createGroupInvitation_
+    Nothing -> createGroupInvitation_
     -- TODO treat the case that the invitation details could've changed
     Just gId -> getGroupInfo db user gId
   where
     getInvitationGroupId_ :: IO (Maybe Int64)
     getInvitationGroupId_ =
       maybeFirstRow fromOnly $
         DB.query db "SELECT group_id FROM groups WHERE inv_queue_info = ? AND user_id = ? LIMIT 1" (connRequest, userId)
-    createGroupInvitation_ :: IO (Either StoreError GroupInfo)
+    createGroupInvitation_ :: ExceptT StoreError IO GroupInfo
     createGroupInvitation_ = do
       let GroupProfile {displayName, fullName, image} = groupProfile
-      withLocalDisplayName db userId displayName $ \localDisplayName -> do
-        currentTs <- getCurrentTime
-        DB.execute
-          db
-          "INSERT INTO group_profiles (display_name, full_name, image, user_id, created_at, updated_at) VALUES (?,?,?,?,?,?)"
-          (displayName, fullName, image, userId, currentTs, currentTs)
-        profileId <- insertedRowId db
+      ExceptT $
+        withLocalDisplayName db userId displayName $ \localDisplayName -> runExceptT $ do
+          currentTs <- liftIO getCurrentTime
+          groupId <- liftIO $ do
+            DB.execute
+              db
+              "INSERT INTO group_profiles (display_name, full_name, image, user_id, created_at, updated_at) VALUES (?,?,?,?,?,?)"
+              (displayName, fullName, image, userId, currentTs, currentTs)
+            profileId <- insertedRowId db
+            DB.execute
+              db
+              "INSERT INTO groups (group_profile_id, local_display_name, inv_queue_info, user_id, created_at, updated_at) VALUES (?,?,?,?,?,?)"
+              (profileId, localDisplayName, connRequest, userId, currentTs, currentTs)
+            insertedRowId db
+          _ <- createContactMemberInv_ db user groupId contact fromMember GCHostMember GSMemInvited IBUnknown fromMemberProfile currentTs
+          membership <- createContactMemberInv_ db user groupId user invitedMember GCUserMember GSMemInvited (IBContact contactId) incognitoProfile currentTs
+          pure GroupInfo {groupId, localDisplayName, groupProfile, membership, membershipIncognito = isJust incognitoProfile, createdAt = currentTs, updatedAt = currentTs}
+
+createContactMemberInv_ :: IsContact a => DB.Connection -> User -> GroupId -> a -> MemberIdRole -> GroupMemberCategory -> GroupMemberStatus -> InvitedBy -> Maybe Profile -> UTCTime -> ExceptT StoreError IO GroupMember
```

Comment:
this asks for some type to pass here, this number of parameters is overkill.

Isn't it GroupInvitation?